lms/db/_session_extension.py

```diff
@@ -0,0 +1,32 @@
+class SessionExtension:
+    """A self returning descriptor to adding functionality to a session.
+
+    To use this, sub-class it and then add your object to the session:
+
+    class CustomSession(Session):
+        my_extension = MyExtension()
+
+    The `name` and `session` attributes will be available to your instance.
+    """
+
+    session = None
+    name = None
+
+    def __set_name__(self, owner, name):
+        # Store the name for more informative errors
+        self.name = name
```

Comment:
I doubt any children will need the name, but what the hoo. It means we can give a nice error message in a few lines if you try to access this as a class attribute.